protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -120,8 +120,10 @@ def _new_project_config(
 
     @staticmethod
     def _validate_project_name(name: str):
-        pattern = r"^[a-zA-Z0-9][a-zA-Z0-9_]*$"
-        if not re.match(pattern, name):
+        # https://github.com/software-mansion/scarb/blob/main/scarb/src/core/package/name.rs#LL42C9
+        pattern = r"^[a-zA-Z_][0-9a-zA-Z_]*$"
+        if not re.match(pattern, name) or name == "_":
```

Comment:
isn't this doable by modifying the regex itself?

Reply:
It is, but as @cptartur noticed [above](https://github.com/software-mansion/protostar/pull/1849#discussion_r1186160961), it becomes less readable

Reply:
Why wouldn't we want to use the regex though? The naming rules described in documentation are clear, the user can always refer to them. Even if they are too lazy to do so, they will probably just use another name. I feel like we are focusing too much on the less important details in this PR.

Reply:
The problem with regexes is that they are not really intuitive e.g. I'm not sure if knowing what `^` or `$` mean is a basic knowledge. And the point of these error messages is to guide the user - too much complexity and they stop being useful at all.

Reply:
I would say let's leave it to @piotmag769 call, it is an unimportant detail not worth four people discussing it. Please improve the error message though